src/utilities/flatten.ts

```diff
@@ -0,0 +1,45 @@
+import { isMoment } from 'moment';
+
+const isBuffer = (obj: any) => {
+  return obj && obj.constructor && typeof obj.constructor.isBuffer === 'function' && obj.constructor.isBuffer(obj);
+};
+
+const keyIdentity = (key: string) => {
+  return key;
+};
+
+export const flatten = (target: Object, opts: any = {}) => {
```

Comment:
Is this from somewhere? And why this instead of `import flattenObject from 'flat';` like in FormManager?

Reply:
Yep this is mostly taken from `flat`'s `flattenObject`. I added an `opts` to handle `Moment` objects (see `isDate` below) because the `flattenObject` was flattening the whole `Moment` object when we just want to keep it as a value, not a series of new key: value pairs. That was messing up all the `datePicker` inputs

Reply:
Ah interesting!

1. Can you add a comment to the code about all this?

Future ideas:
2. We should probably just remove datepicker soon, they are awful. We only use 1 in registry and it would be better as date
3. We could replace this with date-fns and reduce bundle size: https://ant.design/docs/react/replace-moment#Use-date-fns

Reply:
Yeah that makes sense. Would be a good follow-up after ant-4 is fully released